packages/zk-transfer-web/src/index.tsx

```diff
@@ -5,7 +5,14 @@ import "./styles/tailwind.min.css";
 import App from "./App";
 import * as serviceWorker from "./serviceWorker";
 
-ReactDOM.render(<App />, document.getElementById("root"));
+document.addEventListener('DOMContentLoaded', async () => {
+    const { demo } : any = window;
+
+    await demo.clientInit();
+    console.log('Before rendering', demo);
+        
+    ReactDOM.render(<App demo={demo}/>, document.getElementById("root"));
```

Comment:
Perhaps we can call
`ReactDOM.render` before `clientInit` so it doesn't block the page render while the wallet initialization happens? As you noted, it takes a few seconds.

Something like
```
ReactDOM.render(<App demo={demo}/>
await demo.clientInit();
...
// update ReactDOM with new ethereum address somehow, if that's possible?
```

I'll keep thinking of ways to doing the above. It's nice to have fast render, but not urgent (you can work on this last if you have time, I'll also keep thinking of ideas like using React futures or something similar).

Reply:
Done! It now loads asynchronously.

Reply:
\o/ nice work